src/app/dashboard/dashboard.component.ts

```diff
@@ -18,31 +18,25 @@ export class DashboardComponent implements OnInit {
 
     ngOnInit() {
         this.getProjects();
-        this.setPage();
+        this.setPage(this.currentPage);
     }
 
     public getProjects() {
         this.projects = this.projectService.list();
     }
 
     public deleteProject(project) {
-        // special handling for page with single project
-        if (this.pageOfProjects.length == 1 && this.currentPage !==1) {
-            // first switch to prior page else pagination breaks
-            this.currentPage -= 1;
-            this.setPage();
-            // then remove project
-            this.projectService.remove(project);
-            this.getProjects();
-        } else {
-            this.projectService.remove(project);
-            this.getProjects();
-            this.setPage();
+        this.projectService.remove(project);
+        this.getProjects();
+        let page = this.currentPage;
+        if (this.projects.length % this.itemsPerPage === 0) {
```

Comment:
If you have n* this.itemsPerPage +1 projects (i.e, 6, 11, 16..) and delete a project on a full page, this check fails us and sends us to the previous page (which in the case of 6 -> projects, sends up to blankness)

Oddly enough any deleting any project always produces this error on Firefox.